src/openforms/formio/registry.py

```diff
@@ -85,6 +91,7 @@ def normalize(self, component: Component, value: Any) -> Any:
         """
         Given a value from any source, normalize it according to the component rules.
         """
+        assert "type" in component
```

Comment:
Limitations of `TypedDict` in Python < 3.11, because of the `total=False`, there's no guarantee that the key is present and type checkers (rightfully) complain about that. The assertion helps the type checker as it guarantees that the key is indeed present.